jaxrs-clients/src/test/java/com/palantir/remoting1/jaxrs/feignimpl/Java8TestServer.java

```diff
@@ -0,0 +1,225 @@
+            if (Strings.isNullOrEmpty(value)) {
+                return ImmutableMap.of();
+            } else {
+                return ImmutableMap.of(value, value);
+            }
+        }
+
+        @Override
+        public ImmutableMap<String, String> getThrowsNotFound(@Nullable String value) {
+            if (Strings.isNullOrEmpty(value)) {
+                throw new NotFoundException("Not found");
+            } else {
+                return ImmutableMap.of(value, value);
+            }
+        }
+
+        @Override
+        public ImmutableMap<String, String> getThrowsNotAuthorized(@Nullable String value) {
+            if (Strings.isNullOrEmpty(value)) {
+                throw new NotAuthorizedException("Not authorized");
+            } else {
+                return ImmutableMap.of(value, value);
+            }
+        }
+
+        @Override
+        public Optional<ImmutableMap<String, String>> getOptionalThrowsNotAuthorized(@Nullable String value) {
+            if (Strings.isNullOrEmpty(value)) {
+                throw new NotAuthorizedException("Not authorized");
+            } else {
+                return Optional.of(ImmutableMap.of(value, value));
+            }
+        }
+
+        @Override
+        public ImmutableMap<String, String> getThrowsForbidden(@Nullable String value) {
+            if (Strings.isNullOrEmpty(value)) {
+                throw new ForbiddenException("Forbidden");
+            } else {
+                return ImmutableMap.of(value, value);
+            }
+        }
+
+        @Override
+        public Optional<ImmutableMap<String, String>> getOptionalThrowsForbidden(@Nullable String value) {
+            if (Strings.isNullOrEmpty(value)) {
+                throw new ForbiddenException("Forbidden");
+            } else {
+                return Optional.of(ImmutableMap.of(value, value));
+            }
+        }
+
+        @Override
+        public String getString(@Nullable String value) {
+            return value;
+        }
+
+        @Override
+        public InputStream writeInputStream(String bytes) {
+            return new ByteArrayInputStream(bytes.getBytes(StandardCharsets.UTF_8));
+        }
+
+        @Override
+        public String readInputStream(InputStream data) {
+            try {
+                return new String(Util.toByteArray(data), StandardCharsets.UTF_8);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        @Override
+        public Optional<String> getOptionalString(@Nullable String value) {
+            return Optional.ofNullable(value);
+        }
+
+        @Override
+        public Java8ComplexType getJava8ComplexType(Java8ComplexType value) {
+            return value;
+        }
+    }
+
+    @Path("/")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public interface TestService {
```

Comment:
I think it'd be wise to add some test cases for accepting a `QueryParam` that's `Optional`